kyuubi-server/src/main/scala/org/apache/kyuubi/operation/ExecuteStatement.scala

```diff
@@ -120,6 +118,13 @@ class ExecuteStatement(
           val ke = KyuubiSQLException(s"UNKNOWN STATE for $statement")
           setOperationException(ke)
       }
+
+      val appUser = session.asInstanceOf[KyuubiSessionImpl].engine.appUser
+      val sessionManager = session.sessionManager.asInstanceOf[KyuubiSessionManager]
+      sessionManager.credentialsManager.sendCredentialsIfNeeded(
```

Comment:
do we need send token if remote state is terminated ?

Reply:
No matter where in `ExecuteStatement#waitStatementComplete` we put the send method, there is possibility that Statement has been closed terminated when we send credentials.  

Let's make the possibility more obvious.